lib/rack/cache/storage.rb

```diff
@@ -38,6 +38,9 @@ def create_store(type, uri)
         else
           fail "Unknown storage provider: #{uri.to_s}"
         end
+      elsif uri.nil?
+        # hack to support setting entitystore to nil, which doesn't persist request bodies
+        type.const_get(uri.inspect.upcase).resolve(uri)
       else
         # hack in support for passing a Dalli::Client or Memcached object
         # as the storage URI.
```

Comment:
hmm how about we stop doing this complex string mapping and accept any instance,
so this can be `entitystore: Rack::Cache::EntityStore::Noop.new` and we don't need any parsing ?